src/Components/Psychologists/index.js

```diff
@@ -1,9 +1,250 @@
+      from: 1200,
+      to: 1200
+    },
+    thursday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    friday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    sunday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    },
+    saturday: {
+      availability: false,
+      from: 1200,
+      to: 1200
+    }
+  },
+  email: '',
+  firstName: '',
+  lastName: '',
+  password: '',
+  phone: '',
+  username: ''
+};
 
 function Psychologists() {
+  const [psychologists, setPsychologist] = useState([]);
+  const [toggleForm, setToggleForm] = useState(false);
+  const [itemOnEdit, setItemOnEdit] = useState(itemOnEditInitialState);
+  const [isEditing, setIsEditing] = useState(false);
+  const [showAvailabilityModal, setShowAvailabilityModal] = useState(false);
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const [psychologistAvailability, setPsychologistAvailability] = useState({});
+
+  useEffect(() => {
+    getPsychologists();
+    console.log(psychologists);
+  }, []);
+
+  const getPsychologists = () => {
+    fetch(`${process.env.REACT_APP_API}/psychologists`)
+      .then((response) => response.json())
+      .then((response) => {
+        setPsychologist(response.data);
+      })
+      .catch((error) => error);
+  };
+
+  const toggleFormDisplay = () => {
+    setToggleForm(!toggleForm);
+    setItemOnEdit(itemOnEditInitialState);
+  };
+
+  const toggleAvailabilityModal = (e, psychologist) => {
+    if (!showAvailabilityModal) {
+      e.stopPropagation();
+      setPsychologistAvailability(psychologist.availability);
+      setShowAvailabilityModal(!showAvailabilityModal);
+    } else {
+      setShowAvailabilityModal(!showAvailabilityModal);
+    }
+  };
+
+  const toggleConfirmModal = (e, psychologist) => {
+    if (!showConfirmModal) {
+      e.stopPropagation();
+      setItemOnEdit(psychologist);
+      setShowConfirmModal(true);
+    } else {
+      setItemOnEdit(itemOnEditInitialState);
+      setShowConfirmModal(false);
+    }
+  };
+
+  const handleEdit = (psychologist) => {
+    toggleFormDisplay();
+    setItemOnEdit(psychologist);
+    setIsEditing(true);
+  };
+
+  const onChange = (e) => {
+    if (e.target.name.match(/^(firstName|lastName|username|password|email|address)$/)) {
+      setItemOnEdit({
+        ...itemOnEdit,
+        [e.target.name]: e.target.value
+      });
+    } else {
+      const keys = e.target.name.split('.');
+      setItemOnEdit({
+        ...itemOnEdit,
+        availability: {
+          ...itemOnEdit.availability,
+          [keys[0]]: {
+            ...itemOnEdit.availability[keys[0]],
+            [keys[1]]: e.target.value
+          }
+        }
+      });
+    }
+  };
+
+  const handleSubmit = (item, e) => {
+    e.preventDefault();
+
+    // format values before sending to server as input set value type to string
+    const formattedAvailability = Object.keys(item.availability).reduce(
```

Comment:
very complex, use different states for each input. You should keep in mind this function should be understandable to the rest of the team